include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -227,7 +227,7 @@ public :
                 && point_y >= geometry::get<min_corner, 1>(section.bounding_box) - 1
                 && point_y <= geometry::get<max_corner, 1>(section.bounding_box) + 1)
             {
-                for (int i = section.begin_index + 1; i <= section.end_index; i++)
+                for (signed_index_type i = section.begin_index + 1; i <= section.end_index; i++)
```

Comment:
Indeed, also the type of `non_duplicate_index` and of various variables in the sectionalize code.